src/averageAges.js

```diff
@@ -59,8 +77,41 @@ function calculateAverageAgeDiff(people, onlyWithSon) {
   // 2. keep people who have mothers in the array
   // 3. calculate the difference child.born - mother.born
   // 4. return the average value
+  const childrens = people.filter(
+    person =>
+      people.some(mother => mother.name === person.mother)
+        && (onlyWithSon
+          ? isMale(person)
+          : true)
+  );
+
+  return childrens.reduce((sum, child) => {
+    const mother = people.find(
+      person =>
+        child.mother === person.name
+    );
+
+    return sum + (child.born - mother.born);
+  }, 0) / childrens.length;
+}
+
+function getSex(sex) {
+  return person => person.sex === sex;
+}
+
+function getAge(person) {
+  return person.died - person.born;
 }
 
+function getAverageAge(people) {
+  return +(people.reduce((sum, person) =>
+    sum + getAge(person), 0) / people.length
+  ).toFixed(2);
+}
```

Comment:
Now everything is written correctly, but such code is difficult to read and maintain. Let's take it apart a little bit.
```suggestion
function getAverageAge(people) {
  const totalAge = people.reduce((sum, person) => sum + getAge(person), 0);
  return +(totalAge / people.length).toFixed(2);
}
```